main.cpp

```diff
@@ -60,8 +62,55 @@ struct CommandLineOptions {
     QString updateUpdaterVersion;
     bool updateGame;
     bool playNow;
+    QString connectUrl;
 };
 
+static void argParseError(const QString& message)
+{
+    QMessageBox::critical(nullptr, "Unvanquished Updater", message);
+    exit(1);
+}
+
+QString getURLFromPositionalOptions(const QCommandLineParser& parser)
+{
+    QStringList args = parser.positionalArguments();
+    if (args.empty()) {
+        return "";
+    } else if (args.size() > 1) {
+        argParseError("Too many command line arguments.");
+    }
```

Comment:
Is this correct with the --?

Reply:
I tested it.

Reply:
Why does that work? The QT thing make `--` magical?

Reply:
Yes, the Qt arg parser enables `--` handling by default.

Reply:
Ah interesting.